x-pack/libbeat/docker-compose.yml

```diff
@@ -51,6 +48,6 @@ services:
     healthcheck:
       test: ["CMD-SHELL", 'python -c ''import urllib, json; response = urllib.urlopen("http://elastic:changeme@localhost:5601/api/status"); data = json.loads(response.read()); exit(1) if data["status"]["overall"]["state"] != "green" else exit(0);''']
       retries: 1200
-      interval: 1s
+      interval: 5s
```

Comment:
Any reason for this quite long delay and lower interval? This will slow down builds I would assume?

Reply:
Well slow down is actually relative in the healthcheck, if you wait for 5s instead of 1s, it's 4 fewer requests ES has to deal before the next check. Since Travis is not the fastest thing in the world, giving it more time to start is a good compromise. You see that I've only modified theses limit for the `x-pack/libbeat/docker-compose.yml` and not the other files, it's actually because it takes more times to start and recover the security index to allow authentification.

@ruflin if the test suite becomes flaky or unreliable we will have to switch to file-based authentification for ES.

Reply:
I would hope that ES can cope with 1 request per sec ;-) For the start_period, tried to find some good docs but I assume it waits initially 60s? Seems a bit too much.

Overall my goal is to keep the CI time down, as soon as a service is available, tests should start.

Reply:
I think too, but well its Travis its docker, I prefer to give it as much chance as possible, maybe call that I am fed up with that and I would prefer to not whack a mole more :D

The `start_period` is a really bad name, it should be called `grace period`.
When the container starts, it means that health check that fails will not increase the fail count but if the health check succeeds during that period the container can be marked as healthy, when this happen the remaining time of the `start_period` is ignored. So it doesn't really impact that much the startup it just gives a bit more time to be healthy.

Reply:
SGTM, good argument ;-)